### ISSUE TITLE
error in centos7 #include <execution>

### ISSUE THREAD

**Issue by wyfSunflower** [oneapi-src/oneTBB#1160]
use vcpkg install, it can not found header file while ``g++ -ltbb``
build from source use such way:
``git clone oneTBB;cd build;cmake ..;make -j$(nproc); make install``
just import  and link:
``#include <execution>``
``g++  g++ -ltbb``
``In file included from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:3:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: In function ‘void __pstl::__par_backend::__cancel_execution()’:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:70:10: error: ‘tbb::task’ has not been declared
   70 |     tbb::task::self().group()->cancel_group_execution();
      |          ^~~~
In file included from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:3:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: At global scope:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:410:1: error: expected class-name before ‘{’ token
  410 | {
      | ^
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:411:23: error: ‘task’ in namespace ‘tbb’ does not name a type
  411 |     /*override*/ tbb::task*
      |                       ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:434:6: error: ‘task’ in namespace ‘tbb’ does not name a type
  434 | tbb::task*
      |      ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:479:1: error: expected class-name before ‘{’ token
  479 | {
      | ^
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:486:23: error: ‘task’ in namespace ‘tbb’ does not name a type
  486 |     /*override*/ tbb::task*
      |                       ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:516:6: error: ‘task’ in namespace ‘tbb’ does not name a type
  516 | tbb::task*
      |      ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: In lambda function:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:596:24: error: ‘tbb::task’ has not been declared
  596 |             using tbb::task;
      |                        ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:597:13: error: ‘task’ has not been declared
  597 |             task::spawn_root_and_wait(*new (task::allocate_root())
      |             ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:597:45: error: ‘task’ was not declared in this scope; did you mean ‘tbb::detail::d1::task’?
  597 |             task::spawn_root_and_wait(*new (task::allocate_root())
      |                                             ^~~~
      |                                             tbb::detail::d1::task
In file included from /usr/local/include/oneapi/tbb/parallel_for.h:23,
                 from /usr/local/include/tbb/parallel_for.h:17,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:20,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:3:
/usr/local/include/oneapi/tbb/detail/_task.h:214:31: note: ‘tbb::detail::d1::task’ declared here
  214 | class alignas(task_alignment) task : public task_traits {
      |                               ^~~~
In file included from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:3:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: In lambda function:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:634:18: error: ‘tbb::task’ has not been declared
  634 |             tbb::task::spawn_root_and_wait(*new (tbb::task::allocate_root()) _TaskType(
      |                  ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:634:55: error: ‘task’ is not a member of ‘tbb’; did you mean ‘tbb::detail::d1::task’?
  634 |             tbb::task::spawn_root_and_wait(*new (tbb::task::allocate_root()) _TaskType(
      |                                                       ^~~~
In file included from /usr/local/include/oneapi/tbb/parallel_for.h:23,
                 from /usr/local/include/tbb/parallel_for.h:17,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:20,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:3:
/usr/local/include/oneapi/tbb/detail/_task.h:214:31: note: ‘tbb::detail::d1::task’ declared here
  214 | class alignas(task_alignment) task : public task_traits {``

recompile from source again, it throw such error:
``In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/string:52,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/stdexcept:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/array:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/tuple:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/functional:54,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_utils.h:23,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_range_common.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_concurrent_unordered_base.h:24,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/concurrent_unordered_map.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/concurrent_unordered_map.h:17,
                 from /home/oneTBB/test/tbb/test_concurrent_unordered_map.cpp:22:
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h: In instantiation of ‘class std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >’:
/home/oneTBB/test/common/concurrent_associative_common.h:964:29:   required from ‘void test_map_specific_types() [with Checker = UnorderedMapTypesTester]’
/home/oneTBB/test/tbb/test_concurrent_unordered_map.cpp:97:54:   required from here
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:3106:63: error: no class template named ‘rebind’ in ‘class tbb::detail::d1::tbb_allocator<char>’
       typedef typename _Alloc::template rebind<_CharT>::other _CharT_alloc_type;
                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [test/CMakeFiles/test_concurrent_unordered_map.dir/tbb/test_concurrent_unordered_map.cpp.o] Error 1
make[1]: *** [test/CMakeFiles/test_concurrent_unordered_map.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/string:52,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/stdexcept:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/array:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/tuple:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/functional:54,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_utils.h:23,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_range_common.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_concurrent_skip_list.h:25,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/concurrent_map.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/concurrent_map.h:17,
                 from /home/oneTBB/test/tbb/test_concurrent_map.cpp:19:
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h: In instantiation of ‘class std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >’:
/home/oneTBB/test/common/concurrent_associative_common.h:964:29:   required from ‘void test_map_specific_types() [with Checker = OrderedMapTypesTester]’
/home/oneTBB/test/tbb/test_concurrent_map.cpp:90:52:   required from here
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:3106:63: error: no class template named ‘rebind’ in ‘class tbb::detail::d1::tbb_allocator<char>’
       typedef typename _Alloc::template rebind<_CharT>::other _CharT_alloc_type;
                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [test/CMakeFiles/test_concurrent_map.dir/tbb/test_concurrent_map.cpp.o] Error 1
make[1]: *** [test/CMakeFiles/test_concurrent_map.dir/all] Error 2
In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/string:52,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/stdexcept:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/array:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/tuple:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/functional:54,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_utils.h:23,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_range_common.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_concurrent_unordered_base.h:24,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/concurrent_unordered_set.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/concurrent_unordered_set.h:17,
                 from /home/oneTBB/test/tbb/test_concurrent_unordered_set.cpp:22:
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h: In instantiation of ‘class std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >’:
/home/oneTBB/test/tbb/test_concurrent_unordered_set.cpp:72:115:   required from ‘void UnorderedSetTypesTester::check(const std::list<ValueType>&) [with bool DefCtorPresent = true; ValueType = std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >]’
/home/oneTBB/test/common/concurrent_associative_common.h:909:5:   required from ‘void test_set_specific_types() [with Checker = UnorderedSetTypesTester]’
/home/oneTBB/test/tbb/test_concurrent_unordered_set.cpp:78:54:   required from here
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:3106:63: error: no class template named ‘rebind’ in ‘class tbb::detail::d1::tbb_allocator<char>’
       typedef typename _Alloc::template rebind<_CharT>::other _CharT_alloc_type;
                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [test/CMakeFiles/test_concurrent_unordered_set.dir/tbb/test_concurrent_unordered_set.cpp.o] Error 1
make[1]: *** [test/CMakeFiles/test_concurrent_unordered_set.dir/all] Error 2
In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/string:52,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/stdexcept:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/array:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/tuple:39,
                 from /opt/rh/devtoolset-8/root/usr/include/c++/8/functional:54,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_utils.h:23,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_range_common.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/detail/_concurrent_skip_list.h:25,
                 from /home/oneTBB/src/tbb/../../include/tbb/../oneapi/tbb/concurrent_set.h:21,
                 from /home/oneTBB/src/tbb/../../include/tbb/concurrent_set.h:17,
                 from /home/oneTBB/test/tbb/test_concurrent_set.cpp:20:
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h: In instantiation of ‘class std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >’:
/home/oneTBB/test/tbb/test_concurrent_set.cpp:66:67:   required from ‘void OrderedSetTypesTester::check(const std::list<ValueType>&) [with bool DefCtorPresent = true; ValueType = std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >]’
/home/oneTBB/test/common/concurrent_associative_common.h:909:5:   required from ‘void test_set_specific_types() [with Checker = OrderedSetTypesTester]’
/home/oneTBB/test/tbb/test_concurrent_set.cpp:72:52:   required from here
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:3106:63: error: no class template named ‘rebind’ in ‘class tbb::detail::d1::tbb_allocator<char>’
       typedef typename _Alloc::template rebind<_CharT>::other _CharT_alloc_type;
                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [test/CMakeFiles/test_concurrent_set.dir/tbb/test_concurrent_set.cpp.o] Error 1
make[1]: *** [test/CMakeFiles/test_concurrent_set.dir/all] Error 2``

**Comment by aleksei-fedotov** [comment]
Try setting `PSTL_USE_PARALLEL_POLICIES` (if you are using libstdc++ 9) or `_GLIBCXX_USE_TBB_PAR_BACKEND` (if you are using libstdc++ 10) macro to zero before inclusion of the <execution> header file.

**Comment by wyfSunflower** [comment]
> `PSTL_USE_PARALLEL_POLICIES`

```#define PSTL_USE_PARALLEL_POLICIES 0
#include <execution>```
use ``g++ -std=c++17 -lpthread -ltbb z.cc``,it throw such error:
``In file included from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:4:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: In function ‘void __pstl::__par_backend::__cancel_execution()’:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:70:10: error: ‘tbb::task’ has not been declared
   70 |     tbb::task::self().group()->cancel_group_execution();
      |          ^~~~
In file included from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:4:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: At global scope:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:410:1: error: expected class-name before ‘{’ token
  410 | {
      | ^
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:411:23: error: ‘task’ in namespace ‘tbb’ does not name a type
  411 |     /*override*/ tbb::task*
      |                       ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:434:6: error: ‘task’ in namespace ‘tbb’ does not name a type
  434 | tbb::task*
      |      ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:479:1: error: expected class-name before ‘{’ token
  479 | {
      | ^
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:486:23: error: ‘task’ in namespace ‘tbb’ does not name a type
  486 |     /*override*/ tbb::task*
      |                       ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:516:6: error: ‘task’ in namespace ‘tbb’ does not name a type
  516 | tbb::task*
      |      ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: In lambda function:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:596:24: error: ‘tbb::task’ has not been declared
  596 |             using tbb::task;
      |                        ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:597:13: error: ‘task’ has not been declared
  597 |             task::spawn_root_and_wait(*new (task::allocate_root())
      |             ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:597:45: error: ‘task’ was not declared in this scope; did you mean ‘tbb::detail::d1::task’?
  597 |             task::spawn_root_and_wait(*new (task::allocate_root())
      |                                             ^~~~
      |                                             tbb::detail::d1::task
In file included from /usr/local/include/oneapi/tbb/parallel_for.h:23,
                 from /usr/local/include/tbb/parallel_for.h:17,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:20,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:4:
/usr/local/include/oneapi/tbb/detail/_task.h:214:31: note: ‘tbb::detail::d1::task’ declared here
  214 | class alignas(task_alignment) task : public task_traits {
      |                               ^~~~
In file included from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:4:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h: In lambda function:
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:634:18: error: ‘tbb::task’ has not been declared
  634 |             tbb::task::spawn_root_and_wait(*new (tbb::task::allocate_root()) _TaskType(
      |                  ^~~~
/usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:634:55: error: ‘task’ is not a member of ‘tbb’; did you mean ‘tbb::detail::d1::task’?
  634 |             tbb::task::spawn_root_and_wait(*new (tbb::task::allocate_root()) _TaskType(
      |                                                       ^~~~
In file included from /usr/local/include/oneapi/tbb/parallel_for.h:23,
                 from /usr/local/include/tbb/parallel_for.h:17,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend_tbb.h:20,
                 from /usr/local/include/c++/9.3.0/pstl/parallel_backend.h:14,
                 from /usr/local/include/c++/9.3.0/pstl/algorithm_impl.h:25,
                 from /usr/local/include/c++/9.3.0/pstl/glue_execution_defs.h:52,
                 from /usr/local/include/c++/9.3.0/execution:32,
                 from z.cc:4:
/usr/local/include/oneapi/tbb/detail/_task.h:214:31: note: ‘tbb::detail::d1::task’ declared here
  214 | class alignas(task_alignment) task : public task_traits {
      |                               ^~~~``

**Comment by MikeDvorskiy** [comment]
Hello @wyfSunflower ,
There are official instructions to eliminate the mentioned issue:

https://oneapi-src.github.io/oneDPL/introduction.html#restrictions.
https://www.intel.com/content/www/us/en/developer/articles/release-notes/intel-oneapi-threading-building-blocks-release-notes.html

`The use of oneDPL together with the GNU C++ standard library (libstdc++) version 9 or 10 may lead to compilation errors (caused by oneTBB API changes). Using libstdc++ version 9 requires TBB version 2020 for the header file. This may result in compilation errors when using C++17 or C++20 and TBB is not found in the environment, even if its use in oneDPL is switched off. To overcome these issues, include oneDPL header files before the standard C++ header files, or disable parallel algorithms support in the standard library. For more information, please see [Intel® oneAPI Threading Building Blocks (oneTBB) Release Notes](https://www.intel.com/content/www/us/en/developer/articles/release-notes/intel-oneapi-threading-building-blocks-release-notes.html).`

`An application using Parallel STL algorithms in libstdc++ versions 9 and 10 may fail to compile due to incompatible interface changes between earlier versions of Threading Building Blocks (TBB) and oneAPI Threading Building Blocks (oneTBB). Disable support for Parallel STL algorithms by defining PSTL_USE_PARALLEL_POLICIES (in libstdc++ 9) or _GLIBCXX_USE_TBB_PAR_BACKEND (in libstdc++ 10) macro to zero before inclusion of the first standard header file in each translation unit.`
